### DISCUSSION TITLE
New cli syntax for firewalld

### DISCUSSION THREAD

**Discussion by unknowndevQwQ** [firewalld/firewalld#1401] [Q&A]
Currently, the cli syntax of firewalld is extremely verbose. Whether it is a new service or IPSets, many unnecessary hyphens need to be entered, and related attributes cannot be completed in one command (for example, adding a description when creating a new IPSet, adding ports and helpers when creating a new service, etc.). At the same time, it is more appropriate to change commands such as --new-xx-from-file to import.

And when trying to forward ipv6 ports, it was found that it does not support the ipv6 suffix matching method similar to openwrt fw's `suffix/cidr or mask`(be like `::1919:810/-64` or `::1919:810/::ffff:ffff:ffff:ffff`).

**Comment by erig0** [comment]
I agree the CLI syntax is verbose and could use work. Changing it and migrating users is non trivial.

As for the IPv6 issue.. open a bug.

**Reply by unknowndevQwQ** [reply]
> I agree the CLI syntax is verbose and could use work. Changing it and migrating users is non trivial.

I think maybe we can let both coexist, name the new cli something like fwctl or fwdctl, and gradually deprecate the original cli while completely deprecating the xt backend.
